cmake/Codegen.cmake

```diff
@@ -107,26 +111,73 @@ if(INTERN_BUILD_ATEN_OPS)
             list(APPEND _file_compile_flags "-gencode;arch=compute_100a,code=sm_100a")
           endif()
         endif()
+        if("${_arch}" STREQUAL "100f")
+          if(_existing_arch_flags MATCHES ".*compute_100.*")
+            list(APPEND _file_compile_flags "-gencode;arch=compute_100f,code=sm_100f")
+          endif()
+        endif()
         if("${_arch}" STREQUAL "120a")
           if(_existing_arch_flags MATCHES ".*compute_120.*")
             list(APPEND _file_compile_flags "-gencode;arch=compute_120a,code=sm_120a")
           endif()
         endif()
+        if("${_arch}" STREQUAL "120f")
+          if(_existing_arch_flags MATCHES ".*compute_120.*")
+            list(APPEND _file_compile_flags "-gencode;arch=compute_120f,code=sm_120f")
+          endif()
+        endif()
       endforeach()
       list(JOIN _file_compile_flags " " _file_compile_flags)
-
-      set_source_files_properties(${file} PROPERTIES COMPILE_FLAGS "${_file_compile_flags}")
+      get_filename_component(file_we ${file} NAME_WE)
+      set(file_obj "${CMAKE_BINARY_DIR}/caffe2/CMakeFiles/torch_cuda.dir/__/aten/src/ATen/native/cuda/${file_we}.cu.o")
+      set(custom_compile_command "")
+      set(custom_compile_command "${CMAKE_CUDA_COMPILER} -forward-unknown-to-host-compiler -fPIC")
+      list(APPEND custom_compile_command "-O2 --expt-relaxed-constexpr -DNDEBUG")
```

Comment:
I still don't love this, is it not possible to get the existing includes + compile options have 1 source of truth

Reply:
Perhaps you are right, I see if I can re-use `ATen_CUDA_INCLUDE` here:
https://github.com/pytorch/pytorch/blob/9a12bab0d305ffa8ee482126af8e76b9daaeec77/aten/src/ATen/CMakeLists.txt#L636-L637